tcl/linter.tcl

```diff
@@ -320,3 +320,64 @@ proc linter_sql_query_indices {string} {
 
     return [regexp -all -inline -indices $pattern $string]
 }
+
+proc linter_procs_first_line {string} {
+    #| Get the first line of the body of each proc in the string.
+
+    set proc_lines [dict create]
+    set commands [linter_tcl_commands $string]
+
+    foreach command $commands {
+        switch -regexp $command {
+            {^proc} {
+                set proc_name [lindex $command 1]
+                set body_lines [split [string trim [lindex $command 3]] "\n"]
+                dict set proc_lines $proc_name [lindex $body_lines 0]
+            }
+            {^namespace eval} {
```

Comment:
I don't see this tested.
We normally avoid proc definitions in namespace eval in favour of using an explicit namespace in the function name itself.
Did you come across examples of this usage ?

Reply:
Ensembles are typically nested and there is usage throughout a lot of our projects.

Some examples in qcode-tcl:

https://github.com/qcode-software/qcode-tcl/blob/73501d9ccc3c8d68a99379fa7477b8d80bb13baa/tcl/aws_s3_s3_uri_parse.tcl

https://github.com/qcode-software/qcode-tcl/blob/e78bf293be51a162421645c8ae7d9e7e0fd229c5/tcl/nsv_dict.tcl

https://github.com/qcode-software/qcode-tcl/blob/724053f8b4c802a8e368ec13ac8a40ba206d0aaa/tcl/cast.tcl#L282

Reply:
I've added an ensemble to the test to cover that case.

```
tclsh linter.test 
linter.test:	Total	17	Passed	17	Skipped	0	Failed	0
```